sys/include/architecture.h

```diff
@@ -33,6 +33,15 @@
 extern "C" {
 #endif
 
+/**
+ * @brief   Set a debug breakpoint
+ * @param[in] value Index of the breakpoint, depending on the target only a limited
```

Comment:
This will likely be always implemented as an instruction, so ROM space will limit the number of breakpoints. Even on cheap AVRs with debugWire (which only have one hardware breakpoint) an unlimited number of break instructions can be added.

In the Cortex-M implementation this number is ignored by the CPU, but can be used by the debugger e.g. to provide context. Maybe is is better to explain how it is used on Cortex M and that it is ignored on all other arches. (If someone implements it on a new platform that also can provide additional data, this help can be extended).

I would also change the wording "debug breakpoint" and reserve that for the `DEBUG_BREAKPOINT()` macro.

Please add some warning about behavior when no debug probe is connected. I'm almost certain that some Cortex M MCUs occasionally get stuck when no debugger is attached and a break instruction is executed in a state that consumes a lot of power (as if the CPU is still active) and only a hard reset can recover them. This is something that one really wants to avoid in production code intended for battery powered devices.

I would also like to see a `HAVE_ARCHITECTURE_BREAKPOINT` that allows to test for an actuall implementation. I could see something like

```C
    puts("Error, debug me!");
    if (HAVE_ARCHITECTURE_BREAKPOINT) {
        ARCHITECTURE_BREAKPOINT(0);
    }
    else {
        /* busy wait for debugger to attach and break to ease backtrace */
        while (1) {}
    }
```

to bs useful. Maybe actually `DEBUG_BREAKPOINT()` could do just that (except for the `puts()`)?

Reply:
Ah I didn't know that, I just assumed the value was used to select a hardware breakpoint.

`DEBUG_BREAKPOINT()` is only enabled with ´DEVELHELP=1`, so it should be disabled in production code. I can add a warning regardless.